src/utils/jss/ShadowRootPortal.tsx

```diff
@@ -16,65 +14,81 @@ globalThis.getComputedStyle = new Proxy(
             return Reflect.apply(target, thisArg, args)
         },
     },
 )
 Object.assign(globalThis.window || {}, { getComputedStyle })
 
-let proxy: HTMLElement | undefined
+let proxy: ShadowRoot | undefined
+const handler: ProxyHandler<ShadowRoot> = {
+    // ! (1) to make it more like a Document
+    // ! (2) to make it more like an Element
+    get(target, property, handler) {
+        // ! (1) make react move event listeners to shadowroot instead of document
+        if (property === 'ownerDocument') return handler
+
+        // ! (2) if it's a function, use ours
+        const val = Reflect.get(target, property, target)
+        if (typeof val === 'function') return val.bind(target)
+
+        // @ts-ignore
+        // ! (1) document can createElement, createElementNS, etc., so if we don't have such methods, use theirs
+        if (typeof val === 'undefined' && typeof document[property] === 'function')
+            // @ts-ignore
+            return document[property].bind(document)
+
+        // ! (2) if it's not a function, use theirs
+        // @ts-ignore
+        return document.body[property]
+    },
+    set(target, property, value) {
+        return Reflect.set(target, property, value, target)
+    },
+}
 
 export function PortalShadowRoot() {
     if (GetContext() === 'options') return document.body
     if (globalThis.location.hostname === 'localhost') return document.body
-    if (!proxy)
-        proxy = new Proxy(document.body, {
-            get(target, key, receiver) {
-                const value = Reflect.get(target, key)
-                if (typeof value === 'function')
-                    return function(...args: any[]) {
-                        return Reflect.apply(value, shadow, args)
-                    }
-                return value
-            },
-            set(target, key, value, receiver) {
-                return Reflect.set(document.body, key, value, document.body)
-            },
-        })
-    return proxy
+    if (!proxy) proxy = new Proxy(portalShadowRoot, handler)
+    return (proxy as unknown) as Element
 }
 
-{
-    // ? Hack for React, let event go through ShadowDom
-    const hackingEvents = new WeakMap<Event, EventTarget[]>()
+// ? do we still need this?
+// // ? Hack for React, let event go through ShadowDom
+// const shadowRoots = new WeakMap<Node, WeakMap<Event, EventTarget[]>>()
 
-    function hack(eventName: string, shadowRoot: ShadowRoot | Element) {
-        shadowRoot.addEventListener(eventName, (e: Event) => {
-            if (hackingEvents.has(e)) return
-            const path = e.composedPath()
-            // @ts-ignore
-            const e2 = new e.constructor(e.type, e)
-            hackingEvents.set(e2, path)
-            shadowRoot.dispatchEvent(e2)
-            e.stopPropagation()
-            e.stopImmediatePropagation()
-        })
-    }
+// export function hackEvent(eventName: string, shadowRoot: ShadowRoot | Element) {
```

Comment:
Remove it?

Reply:
We don't need to clean our codebase... Leaving something for future reference is better.

Reply:
We have git history for reference